### PR TITLE
[None][feat] Honor SamplingParams.seed on the one-model speculative path

### PR DESCRIPTION
**Pull request by zhaoyangwang-nvidia**
## Description

The one-model speculative sampling calls all took a single engine-wide seed tensor, incremented once per call, so a request's `SamplingParams.seed` was silently ignored on this path: every request in a batch shared one RNG stream, and a run's output depended on how many steps the engine had already taken.

This PR carries the seed per request.

- `SpecMetadata` fills Philox seed/offset buffers in two layouts, one entry per request (`request_seeds` / `request_offsets`) and one per logits row (`seeds` / `offsets`), because the sampling calls consume one or the other. Each call site takes the layout matching the slice it already uses for `temperatures` / `top_ks` / `top_ps`, so the RNG state lines up with the sampling params by construction; `_rng_state_per_request` also covers the block sampler's `repeat_interleave(K)` expansion.
- With a fixed user seed the offset is the only value left that can advance, so it carries the position within a request's stream. Each decoding iteration owns a window of `max_draft_len + 1` offsets: the target sampler (or the rejection kernel, its alternative) takes `base + 0` and draft step `i` takes `base + 1 + i`, where `base = py_decoding_iter * (max_draft_len + 1)`. The draft loop needs a slot per step because it launches the sampler once per step, restarting the kernel's per-row subsequence each time — with a fixed seed and a fixed offset every draft step would otherwise draw identical numbers. A block sampler emits all `K` positions in one launch, so its rows are already separated by the subsequence and share one slot.
- Basing the window on the request's own `py_decoding_iter` rather than a global step counter ties a seeded request's stream to how far it has decoded instead of to when it was scheduled.
- A request that sets no seed gets `DEFAULT_SAMPLING_SEED` (42) rather than the old per-step counter. Its rows stay independent through the kernel's per-row subsequence and its steps through the offset, so sampling stays independent while a run becomes reproducible instead of depending on engine uptime. This changes the token stream unseeded requests see today: the distribution is untouched, but it is not bit-for-bit compatible with the previous behavior.
- The three dynamic-tree sampling calls in `eagle3_dynamic_tree.py` that carried their own seed handling are moved onto the same helpers.

Two limits are worth stating up front. The pinned flashinfer reads only element 0 of the seed/offset tensors, separating rows by `blockIdx.x`, so a batch mixing seeds currently draws from its first row's seed; the per-row values are wired end-to-end and take effect once [flashinfer#2345](https://github.com/flashinfer-ai/flashinfer/pull/2345) lands, and a batch on a single seed is already correct today. Reproducibility is also per decoding position rather than bit-for-bit across batch positions, since rows within one step are separated by `blockIdx.x`, a batch-absolute index.

## Test Coverage

Validated on H200 NVL (SM 9.0).

- Offset window encoding, with `max_draft_len=4`: `base = iter * (max_draft_len + 1)`; target offset equals `base`; draft step `i` equals `base + 1 + i`; windows of adjacent iterations do not overlap; the block sampler's `K` rows share one slot.
- Seed population: per-request seeds land in batch order; unseeded requests get `DEFAULT_SAMPLING_SEED`; mixed batches (some seeded, some not) fill correctly; the offset advances across a request's steps; a `uint64` seed is reinterpreted as an `int64` bit pattern; per-token expansion matches the per-request values.
- End-to-end `test_llama_eagle3_rejection_sampling_modes` passes (4 cases), covering `use_cuda_graph` x `use_dynamic_tree` with `temperature=1.0` / `top_p=0.9` so the non-greedy path is exercised.
- `test_advanced_sampling_mode.py::test_no_topk_matches_full[0.9]` is flaky on this branch **and on the base commit** — five baseline runs produced two failures. It compares two probability distributions at `atol=1e-5` and does not touch any code this PR changes, so it is pre-existing rather than a regression from here.
- Changed-file pre-commit hooks pass.

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- If PR introduces API changes, an appropriate PR label is added - either `api-compatible` or `api-breaking`. For `api-breaking`, include `BREAKING` in the PR title.
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`.

## Dev Engineer Review

- Added per-request Philox seeds and offsets to `SpecMetadata`.
- Added `DEFAULT_SAMPLING_SEED = 42` for requests without an explicit seed.
- Added request-level and logits-row RNG-state layouts.
- Updated token, draft-step, block, rejection, and dynamic-tree sampling paths to use metadata-driven RNG state.
- Added `repeat_interleave(K)` handling for block-sampling state expansion.
- Advanced offset windows per sampling pass and request sequence slot.
- Sized request-level RNG buffers to the actual request count for oversized CUDA graph warmup batches.
- Pinned FlashInfer reads only tensor element 0. Mixed-seed batches remain limited until the referenced FlashInfer change is available.

## QA Engineer Review

- Added CPU tests in `test_rng_window_counter.py` for:
  - Offset allocation across sampling passes.
  - Stale decoding-iteration handling.
  - Sequence-slot-based allocation.
  - Dummy-request handling.
  - Shared counters between live and CUDA-graph metadata copies.
- No corresponding `tests/integration/test_lists/` coverage was identified.
- Verdict: insufficient.

### CODE REVIEW

File: tests/unittest/_torch/speculative/hw_agnostic/test_rng_window_counter.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    assert first != second
+
+
+def test_counter_is_keyed_by_slot_not_batch_position():
+    # Batch composition shifts between iterations; a slot's stream must follow
+    # the slot, not where it happens to sit in the batch.
+    meta = _meta()
+    _offsets(meta, [_request(0), _request(1), _request(2)])
+    # Slot 1 finishes; slot 2 moves to batch position 0.
+    assert _offsets(meta, [_request(2), _request(0)]) == [WINDOW, WINDOW]
```
[the remark this answers is not in the document]
Comment (zhaoyangwang-nvidia) [reply tests/unittest/_torch/speculative/hw_agnostic/test_rng_window_counter.py]: this copy runs before the all-greedy early return that shields the other per-request copies, so an oversized warmup batch reached it; fixed in 5fdcf06688 by sizing with max(self.max_num_requests, len(requests)).

File: tests/unittest/_torch/speculative/hw_agnostic/test_rng_window_counter.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+import types
+
+from tensorrt_llm._torch.speculative.interface import SpecMetadata
+
+MAX_DRAFT_LEN = 3
+WINDOW = MAX_DRAFT_LEN + 1
+
+
+def _meta(max_num_requests=8):
```
[the remark this answers is not in the document]
Comment (zhaoyangwang-nvidia) [reply tests/unittest/_torch/speculative/hw_agnostic/test_rng_window_counter.py]: Confirmed — py_decoding_iter is incremented in _update_requests, which the overlap loop runs after _forward_step, so adjacent batches replayed the same window

File: tensorrt_llm/_torch/speculative/interface.py [hunk 3]
```diff
@@ -655,6 +704,99 @@ class SpecMetadata:
...
+
+        flat_seeds: list[int] = []
+        flat_offsets: list[int] = []
+        for seed, offset, num_tokens in zip(request_seeds, request_offsets,
+                                            num_tokens_per_request):
+            flat_seeds.extend(seed for _ in range(num_tokens))
+            flat_offsets.extend(offset for _ in range(num_tokens))
+
+        # Size by the actual request count as well as max_num_requests: CUDA
+        # graph warmup can pass max_batch_size > max_num_requests, and this
```
Comment (mikeiovine) [inline tensorrt_llm/_torch/speculative/interface.py]: Seems like a bug if max_batch_size > max_num_requests, even during warmup

Comment (zhaoyangwang-nvidia) [reply tensorrt_llm/_torch/speculative/interface.py]: Changed to an assert
